### ISSUE TITLE
self referencing definitions trigger an infinite loop

### ISSUE THREAD

**Issue by jfeltesse-mdsol** [braintree/open_api_parser#8]
If a definition happens to have a property with a reference to itself then doing something like
```
OpenApiParser::Document.new(nil, my_openapi_spec_hash, nil).resolve
```
will crash with `stack level too deep`.

Any ideas on how to handle this?
Looking around it seems the OpenAPI spec is OK with this because the underlying JSON Schema spec allows for such things.

**Comment by drewolson** [comment]
I'm not sure what to do here as we fully expand the specification up front. Perhaps we'd need to add laziness to support this.

**Comment by jfeltesse-mdsol** [comment]
Addressed in PR #10
